pgxn/neon/libpagestore.c

```diff
@@ -98,31 +114,58 @@ CheckPageserverConnstring(char **newval, void **extra, GucSource source)
 static void
 AssignPageserverConnstring(const char *newval, void *extra)
 {
-	if (!PagestoreShmemIsValid())
+	/*
+	 * Only postmaster updates the copy in shared memory.
+	 */
+	if (!PagestoreShmemIsValid() || MyProcPid != PostmasterPid)
```

Comment:
I wonder if this works correctly in single-user mode, when postmaster isn't running at all. I'm not sure if single-user mode works with neon at all, though.

Reply:
I think it is better to replace condition `MyProcPid != PostmasterPid` with `IsUnderPostmaster`. Then it should work also in single-user mode, shouldn't it?